packages/telegram-cloud-storage-stamper/src/index.ts

```diff
@@ -0,0 +1,296 @@
+/// <reference lib="dom" />
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import { TelegramCloudStorageStamperError } from "./errors";
+
+declare global {
+  interface Window {
+    Telegram: any;
+  }
+}
+
+// config when telegram stamper is passed in an api key
+export type TTelegramCloudStorageStamperConfig = {
+  cloudStorageAPIKey?: CloudStorageAPIKey; 
+  cloudStorageKey?: string;
+};
+
+export type CloudStorageAPIKey = {
+  apiPublicKey: string;
+  apiPrivateKey: string;
+};
+
+// Constant for default key name
+const DEFAULT_TURNKEY_CLOUD_STORAGE_KEY = "TURNKEY_API_KEY";
+
+/**
+ * Stamper to use within a `TurnkeyClient`
+ */
+export default class TelegramCloudStorageStamper {
+  // This stamper uses a typical apikey stamper under the hood and abstracts away the storage of the actual api keys
+  stamper?: ApiKeyStamper | undefined;
+
+  // the constructor checks if were in a telegram mini app context, and then gives the developer an interface into looking into CloudStorage even without having an API key attached
+  constructor() {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+  }
```

Comment:
is this useful in practice? If this feature detection is useful I'd rather export it as a standalone function ("supportsCloudStorage"). Using this interface means writing:
```ts
try {
    // this actually just checks for local storage, trust me
    // (or go read Turnkey's docs)
    new TelegramCloudStorageStamper()
} catch (e) {
    // abort case
}
// now we're good, cloud storage is supported (look at the try/catch above)
```

Compare this with:
```ts
if (supportsCloudStorage()) {
    // abort
} else {
    // we're good
}
```
No need for extra code comments, the intent becomes very crisp.

Reply:
Actually, I now see what you mean. The goal of this isn't to do feature detection, the goal is to act as a storage wrapper. So the code would be:
```ts
stamper = new TelegramCloudStorageStamper()
await stamper.getItem("MY_KEY")
await stamper.getItem("OTHER_KEY", "VALUE")
```